Zend/zend_builtin_functions.stub.php

```diff
@@ -6,23 +6,24 @@ function zend_version(): string {}
 
 function func_num_args(): int {}
 
-function func_get_arg(int $arg_num): mixed {}
+function func_get_arg(int $argument_number): mixed {}
 
 function func_get_args(): array {}
 
-function strlen(string $str): int {}
+function strlen(string $string): int {}
 
-function strcmp(string $str1, string $str2): int {}
+function strcmp(string $string1, string $string2): int {}
 
-function strncmp(string $str1, string $str2, int $len): int {}
+function strncmp(string $string1, string $string2, int $length): int {}
 
-function strcasecmp(string $str1, string $str2): int {}
+function strcasecmp(string $string1, string $string2): int {}
 
-function strncasecmp(string $str1, string $str2, int $len): int {}
+function strncasecmp(string $string1, string $string2, int $length): int {}
 
-function error_reporting($new_error_level = UNKNOWN): int {}
+function error_reporting(?int $error_level = null): int {}
 
-function define(string $constant_name, $value, bool $case_insensitive = false): bool {}
+/** @param bool|int|float|string|array|resource|null $value */
```

Comment:
I'd prefer "mixed" here.

Reply:
I think `mixed` shouldn't be used as a parameter unless all the types are really accepted, because they will be surprised if they receive a `TypeError`. It's not the case for a mixed return type, since at worst, users will check for a specific type in vain, but no error happens. Or do you have another reason of this preference I'm not aware of (e.g. future support for object constant value)?

Reply:
Yes, I see this more as "objects are not yet supported here" than "define() fundamentally cannot accept objects".

Reply:
OK, I've just realized that you mean to change the PHPDoc. That's easy-peasy.